app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -132,6 +137,7 @@ class SmartTransactionHook {
     const useRegularTransactionSubmit = { transactionHash: undefined };
     if (
       !this.#isSmartTransaction ||
+      this.#isHardwareWallet ||
```

Comment:
As you can see, HW wallets are not allowed to proceed in this hook, which is used for non-swaps transactions. If a swap transaction is sent, it doesn't use this hook yet, so HW wallet usage will be enabled with STX.

Reply:
Hi Dan, I'm a bit confused about how hardware wallet STXs work. Could you explain how they operate if they don’t go through this hook code path?

Reply:
Sure. Swaps check if smart transactions [are allowed](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/swaps/prepare-swap-page/review-quote.js#L1050) before submit and if yes, it submits a swap as a smart transaction. It doesn't use this hook.

Part of the check is the `getSmartTransactionsEnabled` [function](https://github.com/MetaMask/metamask-extension/blob/develop/shared/modules/selectors/smart-transactions.ts#L102), which calls `accountSupportsSmartTx`. Before this PR `getSmartTransactionsEnabled` returned false if there was a HW wallet detected. The HW wallet check from there was removed in this PR, which means `getSmartTransactionsEnabled` can return true and a swap will be submitted as a smart transaction in that case. A user would see UI for confirming a transaction(s) on their HW device and once confirmed, they will be redirected to the STX status page.

Reply:
Dan, Thank you for the explanation!